src/plots/cartesian/axes.js

```diff
@@ -2467,36 +2472,80 @@ axes.getTickSigns = function(ax) {
  *  - {fn} l2p
  * @return {fn} function of calcTicks items
  */
-axes.makeTransFn = function(ax) {
-    var axLetter = ax._id.charAt(0);
-    var offset = ax._offset;
-    return axLetter === 'x' ?
-        function(d) { return strTranslate(offset + ax.l2p(d.x), 0); } :
-        function(d) { return strTranslate(0, offset + ax.l2p(d.x)); };
+axes.makeTransTickFn = function(ax) {
+    return ax._id.charAt(0) === 'x' ?
+        function(d) { return strTranslate(ax._offset + ax.l2p(d.x), 0); } :
+        function(d) { return strTranslate(0, ax._offset + ax.l2p(d.x)); };
 };
 
-axes.makeTransPeriodFn = function(ax) {
-    var axLetter = ax._id.charAt(0);
-    var offset = ax._offset;
-    return axLetter === 'x' ?
+axes.makeTransTickLabelFn = function(ax) {
+    var uv = getTickLabelUV(ax);
+    var u = uv[0];
+    var v = uv[1];
+
+    return ax._id.charAt(0) === 'x' ?
         function(d) {
             return strTranslate(
-                offset + ax.l2p(getPeriodX(d)),
-                0
+                u + ax._offset + ax.l2p(getPosX(d)),
+                v
             );
         } :
         function(d) {
             return strTranslate(
-                0,
-                offset + ax.l2p(getPeriodX(d))
+                v,
+                u + ax._offset + ax.l2p(getPosX(d))
             );
         };
 };
 
-function getPeriodX(d) {
+function getPosX(d) {
     return d.periodX !== undefined ? d.periodX : d.x;
 }
 
+function getTickLabelUV(ax) {
```

Comment:
minor point - u & v don't mean much to me. The function makes sense, just might be worth a comment like "u is a shift along the axis, v is a shift perpendicular to the axis" for future reference.

Reply:
Good call. Added in https://github.com/plotly/plotly.js/pull/5275/commits/40c09b78daba4ddabb2ff4bbed9b2536783bae29.